include/seqan3/search/algorithm/detail/search_trivial.hpp

```diff
@@ -0,0 +1,210 @@
+                        error_left2.total -= delta;
+                        error_left2.substitution -= delta;
+
+                        if constexpr (abort_on_hit)
+                        {
+                            if (_search_trivial<abort_on_hit>(it, query, query_pos + 1, error_left2, delegate))
+                                return true;
+                        }
+                        else
+                        {
+                            _search_trivial<abort_on_hit>(it, query, query_pos + 1, error_left2, delegate);
+                        }
+                    }
+
+                    // Deletion
+                    if (query_pos > 0) // do not allow deletions at the beginning of the query
+                    {
+                        // Match
+                        if (error_left.substitution == 0 && it.last_char() == query[query_pos])
+                        {
+                            if constexpr (abort_on_hit)
+                            {
+                                if (_search_trivial<abort_on_hit>(it, query, query_pos + 1, error_left, delegate))
+                                    return true;
+                            }
+                            else
+                            {
+                                _search_trivial<abort_on_hit>(it, query, query_pos + 1, error_left, delegate);
+                            }
+                        }
+
+                        if (error_left.deletion > 0)
+                        {
+                            search_params error_left2{error_left};
+                            error_left2.total--;
+                            error_left2.deletion--;
+
+                            // do not allow deletion in the next step
+                            if constexpr (abort_on_hit)
+                            {
+                                if (_search_trivial<abort_on_hit>(it, query, query_pos, error_left2, delegate))
+                                    return true;
+                            }
+                            else
+                            {
+                                _search_trivial<abort_on_hit>(it, query, query_pos, error_left2, delegate);
+                            }
+                        }
+                    }
+                }
+                while (it.cycle_back());
+            }
+            else
+            {
+                // Match
+                if (it.extend_right(query[query_pos]))
+                {
+                    if constexpr (abort_on_hit)
+                    {
+                        if (_search_trivial<abort_on_hit>(it, query, query_pos + 1, error_left, delegate))
+                            return true;
+                    }
+                    else
+                    {
+                        _search_trivial<abort_on_hit>(it, query, query_pos + 1, error_left, delegate);
+                    }
+                }
+            }
```

Comment:
the whole code block spans so many lines of code just because of the many braces. Can we not remove the braces for the inner constexpr if blocks?

Reply:
I'd leave it like this since this file will be completely removed from the library and replaced by search schemes in a couple of weeks anyway.